src/tracing.ts

```diff
@@ -68,3 +75,59 @@ export function startTracing(opts: Partial<Options> = {}): void {
   // register global provider
   trace.setGlobalTracerProvider(provider);
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function configureHttpInstrumentation(instrumentation: any, options: Options) {
+  if (!options.serverTimingEnabled) {
+    return;
+  }
+
+  if (
+    typeof instrumentation['setConfig'] !== 'function' ||
+    typeof instrumentation['_getConfig'] !== 'function'
+  ) {
+    return;
+  }
+
+  const responseHook: HttpResponseCustomAttributeFunction = (
+    span,
+    response
+  ) => {
+    if (response instanceof ServerResponse) {
+      const { traceId, spanId } = span.context();
+      response.setHeader('Access-Control-Expose-Headers', 'Server-Timing');
+      response.setHeader(
+        'Server-Timing',
+        `traceparent;desc="00-${traceId}-${spanId}-01"`
+      );
+    }
+  };
+
+  let config = instrumentation._getConfig() as HttpInstrumentationConfig;
```

Comment:
can this result in runtime error? can we add guards in case?

Reply:
I guess only if the logic of `getConfig` changes (e.g. adding required arguments), currently it only returns an existing dictionary

Reply:
OK. As long as we have a test case that detect such breaking I think we should be good.

Reply:
Should be covered by tests in `test/servertiming.test.ts`